salt/client/ssh/__init__.py

```diff
@@ -946,6 +946,20 @@ def run_wfunc(self):
         self.wfuncs = salt.loader.ssh_wrapper(opts, wrapper, self.context)
         wrapper.wfuncs = self.wfuncs
 
+        # If salt-thin is set to be wiped, a 'test.ping' command is executed
+        # after running the wfunc in order to trigger the salt-thin cleanup
+        # done by SHIM on the client
+        if 'ssh_wipe' in self.opts:
+            single = Single(self.opts,
+                            "test.ping",
```

Comment:
I personally not the fan of this approach. Either create a single point of "wipe out" and react as wipe out, or fix _every function_. What you do here is a workaround where "some functions do call it, so let call those functions" thing.

Instead I would do something like:
```python
if 'ssh_wipe' in self.opts:  # "ssh_wipe" should be a constant!
    try:
        wipe_here_the_thing_now()
    except WhateverValidExceptionClassOfTheDay as ex:
        log.error(............)
```